static/app/utils/replays/countDomNodes.tsx

```diff
@@ -0,0 +1,108 @@
+import {Replayer} from '@sentry-internal/rrweb';
+
+import type {RecordingFrame} from 'sentry/utils/replays/types';
+
+export type DomNodeChartDatapoint = {
+  count: number;
+  endTimestampMs: number;
+  startTimestampMs: number;
+  timestampMs: number;
+};
+
+type Args = {
+  frames: RecordingFrame[] | undefined;
+  rrwebEvents: RecordingFrame[] | undefined;
+  startTimestampMs: number;
+};
+
+export default function countDomNodes({
+  frames = [],
+  rrwebEvents,
+  startTimestampMs,
+}: Args): Promise<DomNodeChartDatapoint[]> {
+  return new Promise(resolve => {
+    const datapoints = new Map<RecordingFrame, DomNodeChartDatapoint>();
+    const player = createPlayer(rrwebEvents);
+
+    const nextFrame = (function () {
+      let i = 0;
+      const len = frames.length;
+      // how many frames we look at depends on the number of total frames
+      return () => frames[(i += Math.max(Math.round(len * 0.007), 1))];
```

Comment:
What's the reasoning behind the `len * .007` logic?

Reply:
The frames list is so long and contains data from very similar timestamps, so we want to step in chunks to avoid looking at every single frame -- larger arrays in bigger chunks, smaller arrays in smaller chunks

Reply:
I picked 0.007 arbitrarily because it seemed like a good multiplier to have data load not _too_ slowly, but it also leads to a pretty detailed/smooth chart